plugins/typo_squatter/komand_typo_squatter/util/utils.py

```diff
@@ -1,14 +1,49 @@
+from insightconnect_plugin_runtime.exceptions import PluginException
+import subprocess  # nosec
+import validators
+import json
 import re
-import tqdm
-import entropy
+import math
 from tld import get_tld
 from Levenshtein import distance
-from termcolor import colored, cprint
-
 from .suspicious import keywords, tlds
 
 
-def score_domain(domain):
+def check_for_squatters(domain: str, flag: str, logger) -> list:
+    if not validators.domain(domain):
+        raise PluginException(
+            cause="Invalid domain provided.", assistance="Please provide a valid domain and try again."
+        )
+    cmd = f"dnstwist {flag} -f json {domain}" if flag else f"dnstwist -f json {domain}"
+    logger.info(f"Running command: {cmd}")
+    results = subprocess.run(cmd.split(" "), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+    error = results.stderr.decode()
+    if error:
+        if "unrecognized arguments" in error:
+            raise PluginException(
+                cause="Invalid flag provided.", assistance="Please provide a valid flag and try again.", data=error
+            )
+        else:
+            raise PluginException(
+                cause=f"An error occured while executing the command: {cmd}.",
+                assistance="Please try again and contact support if the problem persists.",
+                data=error,
+            )
+    js = json.loads(results.stdout.decode().replace("\\n", ""))
+    for _, item in enumerate(js):
+        js[_]["phishing_score"] = score_domain(item.get("domain-name"))
+
+    return js
+
+
+def entropy(string: str) -> float:
+    """Calculates the Shannon entropy of a string"""
```

Comment:
This should be a tested function if it was written by hand. It is very complex and not intuitive to the reader. If you got this from an online source, link it in a comment so we can reference it as the reviewer

Reply:
I now see you tested it below. Still, a link to your source would be amazing!

Reply:
I got this method from the link provided in the help.md.

https://github.com/x0rz/phishing_catcher/blob/master/catch_phishing.py

Reply:
```suggestion
    """
    Calculates the Shannon entropy of a string
    Original code: https://github.com/x0rz/phishing_catcher/blob/master/catch_phishing.py
    """
```